In [2]:
import numpy as np
import pandas as pd
import math
from scipy import stats


In [3]:
property_df=pd.read_csv(r"C:\Users\91981\Documents\Parul\Hero Vired Course\python\Assingment_ Python for statistics\property.csv")

In [4]:
property_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  float64
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

In [5]:
property_df.describe()

Rooms         Price      Distance      Postcode      Bedroom2  \
count  13580.000000  1.358000e+04  13580.000000  13580.000000  13580.000000   
mean       2.937997  1.075684e+06     10.137776   3105.301915      2.914728   
std        0.955748  6.393107e+05      5.868725     90.676964      0.965921   
min        1.000000  8.500000e+04      0.000000   3000.000000      0.000000   
25%        2.000000  6.500000e+05      6.100000   3044.000000      2.000000   
50%        3.000000  9.030000e+05      9.200000   3084.000000      3.000000   
75%        3.000000  1.330000e+06     13.000000   3148.000000      3.000000   
max       10.000000  9.000000e+06     48.100000   3977.000000     20.000000   

           Bathroom           Car       Landsize  BuildingArea    YearBuilt  \
count  13580.000000  13518.000000   13580.000000   7130.000000  8205.000000   
mean       1.534242      1.610075     558.416127    151.967650  1964.684217   
std        0.691712      0.962634    3990.669241    541.014538    37.273762   
min        0.000000      0.000000       0.000000      0.000000  1196.000000   
25%        1.000000      1.000000     177.000000     93.000000  1940.000000   
50%        1.000000      2.000000     440.000000    126.000000  1970.000000   
75%        2.000000      2.000000     651.000000    174.000000  1999.000000   
max        8.000000     10.000000  433014.000000  44515.000000  2018.000000   

          Lattitude    Longtitude  Propertycount  
count  13580.000000  13580.000000   13580.000000  
mean     -37.809203    144.995216    7454.417378  
std        0.079260      0.103916    4378.581772  
min      -38.182550    144.431810     249.000000  
25%      -37.856822    144.929600    4380.000000  
50%      -37.802355    145.000100    6555.000000  
75%      -37.756400    145.058305   10331.000000  
max      -37.408530    145.526350   21650.000000

In [6]:
property_df['Price'].value_counts()

Price
1100000.0    113
1300000.0    109
800000.0     109
650000.0     109
600000.0     104
            ... 
3778000.0      1
365500.0       1
791100.0       1
1631888.0      1
2562000.0      1
Name: count, Length: 2204, dtype: int64

## Q1: We have been asked to check whether mean of the price is $800,000 or higher, hence we should apply the Z test 

## H0: Mean is 800,000
H1: Mean is greater than 800,000 

In [7]:
property_df['Price'].mean()

np.float64(1075684.079455081)

In [9]:
x_bar = 800000 #Sample Mean
mu = property_df['Price'].mean() #Population Mean
sigma = 6393.1 #Sample Standard Deviation
n = 13580 #Number of samples

In [10]:
z = (x_bar - mu)/(sigma/math.sqrt(n))

In [11]:
z

np.float64(-5025.162080569881)

In [14]:
p = (1 - stats.norm.cdf(z))

In [15]:
p

np.float64(1.0)

In [16]:
alpha=0.05 # Since significance level is said to be 5%

In [17]:
if p <= alpha:
    print("Reject H0")
else:
    print("Fail to reject H0")

Fail to reject H0


## This means that the mean is not $800,000 - the mean is greater than $800,000. 

## Q2. We need to check if the mean of prices of Summer 2016 = mean of prices of Winter 2016- hence we should apply the T test for independent samples- because the prices of Summer and WInter are independent of each other. 

## H0: Mean of Summer 2016=Mean of Winter 2016
## H1: Mean of Summer 2016 does not equal Mean of Winter 2016

In [19]:
property_df.head()

Suburb           Address  Rooms Type      Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB  Nelson   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
1  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
2  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
3  4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0           NaN   
4  4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN        Yarra  -37.7996    144.9984  Northern Metropolitan   
1     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
2     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   
3        NaN        Yarra  -37.7969    144.9969  Northern Metropolitan   
4     2014.0        Yarra  -37.8072    144.9941  Northern Metropolitan   

  Propertycount  
0        4019.0  
1        4019.0  
2        4019.0  
3        4019.0  
4        4019.0  

[5 rows x 21 columns]

In [22]:
# Convert Date column to datetime (format m/dd/yyyy)
property_df["Date"] = pd.to_datetime(property_df["Date"], format="%d/%m/%Y")

In [23]:
# Define date ranges for winter
range1_start = "2016-01-01"
range1_end   = "2016-03-31"

range2_start = "2016-10-01"
range2_end   = "2016-12-31"

In [24]:
winter = (
    (property_df["Date"].between(range1_start, range1_end)) |
    (property_df["Date"].between(range2_start, range2_end))
)


In [25]:
winter_prices = property_df.loc[winter, "Price"].to_numpy()

In [26]:
# Define date ranges for summer
range3_start = "2016-04-01"
range3_end   = "2016-09-30"

In [27]:
summer = (
    property_df["Date"].between(range3_start, range3_end) 
)

In [28]:
summer_prices=property_df.loc[summer, "Price"].to_numpy()

In [30]:
t_stat,p_value = stats.ttest_ind(winter_prices,summer_prices)

In [31]:
t_stat

np.float64(4.043386317851058)

In [32]:
p_value

np.float64(5.3309767667631686e-05)

In [33]:
if p_value <= 0.05:
    print("Reject H0: Means average summer and average winter prices are different")
else:
    print("Fail to reject H0: Means average summer price and average winter price are equal")

Reject H0: Means average summer and average winter prices are different


In [ ]:
## Q3. For Abbortford, we have to figure out whether the probability of 

In [35]:
property_df['Car'].fillna(value= property_df['Car'].median(), inplace=True) ## Since in the info function right at top we saw that "Car column has only 13518 values and not 13580 values.- showing some discrepancy

C:\Users\91981\AppData\Local\Temp\ipykernel_21468\153794711.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  property_df['Car'].fillna(value= property_df['Car'].median(), inplace=True) ## Since in the info function right at top we saw that "Car column has only 13518 values and not 13580 values.- showing some discrepancy


In [36]:
abbotsford_df = property_df[property_df["Suburb"] == "Abbotsford"]

In [37]:
p = (abbotsford_df["Car"] == 0).mean()

In [38]:
from scipy.stats import binom

In [39]:
# Number of trials (n) and number of successes (k)
n = 10
k = 3

In [40]:
# Binomial probability: P(X = 3)
prob_3 = binom.pmf(k, n, p)

In [44]:
print("Probability p (Property sold has no parking space in Abbotsford):", np.round(p,3))
print("P(X = 3 out of 10):", np.round(prob_3,3))

Probability p (Property sold has no parking space in Abbotsford): 0.268
P(X = 3 out of 10): 0.26


## Q4. Probability of finding a property with 3 rooms in Abbotsford

In [45]:
prob_3_rooms = (abbotsford_df["Rooms"] == 3).mean()

In [46]:
np.round(prob_3_rooms,3)

np.float64(0.357)

#The probability of finding a property with 3 rooms in Abbotsford is 35.7%

## Q5: Probability of finding a property with 2 bathrooms in Abbotsford

In [47]:
prob_2_bathrooms = (abbotsford_df["Bathroom"] == 2).mean()

In [49]:
np.round(prob_2_bathrooms,3)

np.float64(0.339)

#The probability of finding a property with 2 bathrooms in Abbotsford is 33.9%